photutils/isophote/tests/test_isophote.py

```diff
@@ -59,11 +59,13 @@ def test_fit(self):
         assert iso.tflux_c <= 2.025E6
         assert iso.tflux_c >= 2.022E6
 
-        # deviations from perfect ellipticity
-        assert abs(iso.a3) <= 0.01
-        assert abs(iso.b3) <= 0.01
-        assert abs(iso.a4) <= 0.01
-        assert abs(iso.b4) <= 0.01
+        # deviations from perfect ellipticity. Note
+        # that sometimes a None covariance can be
+        # generated by scipy.optimize.leastsq
+        assert iso.a3 is None or abs(iso.a3) <= 0.01
+        assert iso.a3 is None or abs(iso.b3) <= 0.01
+        assert iso.a3 is None or abs(iso.a4) <= 0.01
+        assert iso.a3 is None or abs(iso.b4) <= 0.01
```

Comment:
The first conditionals here all copy `iso.a3`, but should be `b3`, `a4`, `b4`.  The tests are still failing because of this.